src/dev/flang/be/jvm/classfile/Expr.java

```diff
@@ -309,8 +619,11 @@ public void code(ClassFile.ByteCodeWriter ba, ClassFile cf)
    * create invokespecial bytecode to call given class, name and descr producing
    * void result type.
    */
-  public static Expr invokeSpecial(String cls, String name, String descr)
+  public static Expr invokeSpecial(String cls, String name, String descr, int argCount)
```

Comment:
Instead of adding `argCount` as an argument here, try `ClassFileConstants.slotCountForArgs(descr)`.